contracts/dao/PancakeStaking.sol

```diff
@@ -188,4 +207,33 @@ contract PancakeStaking is OwnableUpgradeable, ReentrancyGuardUpgradeable {
     function setHarvestTimeGap(uint256 _harvestTimeGap) external onlyOwner {
         harvestTimeGap = _harvestTimeGap;
     }
+
+    /**
+      * @dev set emergency mode
+      * @param _emergencyMode emergency mode
+      */
+    function setEmergencyMode(bool _emergencyMode) external onlyOwner {
+        emergencyMode = _emergencyMode;
+        emit SetEmergencyMode(_emergencyMode);
+    }
+
+    /**
+      * @dev emergency withdraw all lp tokens from farming contract given the list of lp token addresses
+      * @param lpTokens the list of lp token addresses
+      */
+    function emergencyWithdraw(address[] memory lpTokens) external onlyOwner nonReentrant {
+        for (uint256 i = 0; i < lpTokens.length; ++i) {
+            address lpToken = lpTokens[i];
+            Pool memory poolInfo = pools[lpToken];
+            if (!poolInfo.isActive) {
+                continue; // Skip inactive pool
+            }
+            uint256 lpAmount = IERC20(poolInfo.lpToken).balanceOf(address(this));
+            IV2Wrapper(poolInfo.poolAddress).emergencyWithdraw();
+            lpAmount = IERC20(poolInfo.lpToken).balanceOf(address(this)) - lpAmount;
+
+            emit EmergencyWithdraw(poolInfo.poolAddress, lpAmount);
+        }
+        emergencyMode = false;
```

Comment:
Add an event here

Reply:
done in 49ec5670cb578e687c2bd3d9cd4fabf31f7d2863